packages/issuer-sdk-js/src/internal.ts

```diff
@@ -10,8 +10,8 @@ export function ensureEntityId<T extends { id?: string }>(entity: T): { id: stri
 }
 
 // biome-ignore lint/suspicious/noExplicitAny: Using `any` is fine over here as we don't need to be strict about types being passed in.
-export function createActionInput(params: Record<string, any>): Utils.ActionInput {
-  const payload = ensureEntityId(params);
+export function createActionInput(params: Record<string, any>, idNeeded = true): Utils.ActionInput {
+  const payload = idNeeded ? params : ensureEntityId(params);
```

Comment:
Why did we end up doing this? `ensureEntityId` already doesn't override the `id` field if it's present, right?

Reply:
We need to prepare request payload without id [here](https://github.com/idos-network/idos-sdk-js/pull/512/files#diff-3a869b1ce36c91f549b25277f8814807bf9a41f3c6287c46a8ec8c5ee5737e55R173)

Reply:
All done! Check [84340f4](https://github.com/idos-network/idos-sdk-js/pull/512/commits/84340f4ef8880379ea525a8459e074f0dad42332).